### PR TITLE
treewide: remove isNull where it decreases readability and code understanding

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by infinisil — CHANGES_REQUESTED** [review]
:-1: on this. There's general agreement that it shouldn't be deprecated in https://github.com/NixOS/nix/issues/8264

Edit: Now opened https://github.com/NixOS/nix/pull/9645 to undeprecate it

**Comment by SuperSandro2000** [comment]
I've changed the PR to only remove it where it decreases readability. Especially ``!isNull`` is in my opinion the same category as stacking multiple with: you should not do that.

**Comment by infinisil** [comment]
Out of all the bad code written in Nixpkgs, this is _very_ far down the list imo. I don't think it's worth doing this. Especially because people will just continue using it regardless. And while this _could_ be automated, it's absolutely not worth the trouble. So really, let's just not worry about this.

**Comment by SuperSandro2000** [comment]
So we no longer want to fix bad code because people could write new bad code?

File: pkgs/development/beam-modules/build-mix.nix [hunk 1]
```diff
@@ -61,7 +61,7 @@ let
       runHook preConfigure
 
       ${./mix-configure-hook.sh}
-      ${lib.optionalString (!isNull appConfigPath)
```
Comment (SuperSandro2000) [inline pkgs/development/beam-modules/build-mix.nix:64]: Especially expression like this one shouldn't go through review as they complicate the code a lot.

**Pull request closed without merging**